### PR TITLE
PLAT-95788: Update LabeledIconButton to use Button for icon

### PR DESCRIPTION
**Pull request by JayCanuck**
Switches from `IconButtonBase` to `ButtonBase` as the foundation for `LabeledIconButton`.

Additionally supports the forwarding of the breadth of props for Button, to allow such features as Spinners in LabeledIconButtons, etc..

Enact-DCO-1.0-Signed-off-by: Jason Robitaille <jason.robitaille@lge.com>

### CODE REVIEW

File: LabeledIconButton/LabeledIconButton.js [hunk 1]
```diff
@@ -76,11 +97,29 @@ const LabeledIconButtonBase = kind({
...
 			icon: (
-				<Button pressed={pressed} selected={selected} spriteCount={spriteCount}>{icon}</Button>
+				<Button
+					backgroundOpacity={backgroundOpacity}
+					icon={icon}
+					iconComponent={iconComponent}
+					highlighted={highlighted}
+					pressed={pressed}
+					selected={selected}
+					spriteCount={spriteCount}
```
Comment (ryanjduffy) [inline LabeledIconButton/LabeledIconButton.js]: Is this prop gone now? I'm seeing warnings about it and can't find references any longer.

Comment (Djspaceg) [reply LabeledIconButton/LabeledIconButton.js]: This prop needs to be restored onto Button and sent along to the inner Icon component. I think we may need to update enact/ui to properly add this.

Comment (ryanjduffy) [reply LabeledIconButton/LabeledIconButton.js]: How do you want to proceed for this PR?

Comment (Djspaceg) [reply LabeledIconButton/LabeledIconButton.js]: I will spend a few minutes and see if I can make it joyful again.

Comment (Djspaceg) [reply LabeledIconButton/LabeledIconButton.js]: Let's ignore this for now and address it later. I'm not having any luck fixing it efficiently.

**Pull request merged**